### DISCUSSION TITLE
How to change this "None" text ?

### DISCUSSION THREAD

**Discussion by p1-dta** [evennia/evennia#2416] [Evennia Questions] [answered]
Hi,

I'm trying to translate the index of the website, but I am stuck with this:

![image](https://user-images.githubusercontent.com/9381120/118412379-e4cfb080-b699-11eb-812f-81082057edc7.png)

I want to change the "None" to "Aucun", but I failed to find how to do it.
Maybe the answer is in the doc or in de discussions, but I don't know which keywords used to find those solutions.

```html
        <p>
          Parmis eux, <strong>{{num_accounts_registered_recent}}</strong> ont
          été créé{{num_accounts_registered_recent|pluralize}} cette semaine et
          <strong>{{num_accounts_connected_recent}}</strong> se sont connectés
          dans les 7 derniers jours.
        </p>
```

Thanks

**Comment by Griatch** [comment]
~Not sure, but you could probably use a custom filter/tag inserted something like `{{ num_accounts_registered_recent | tofrench }}` to translate that non-result to `Aucun`. Check out https://docs.djangoproject.com/en/3.2/howto/custom-template-tags/ - maybe that can be of help.~

**Reply by Griatch** [reply]
Actually, it's simpler. This is a part of the index view's `_gamestats`: https://github.com/evennia/evennia/blob/master/evennia/web/website/views.py#L45  If you override the index you can change this to `aucun` instead.

**Reply by p1-dta** [reply]
Can I change it without having to modify Evennia sources ?

**Reply by Griatch** [reply]
Yes of course. You can override any default view by repointing the mygame/web/urls.py for the index to your own, modified view. Check out evennia/urls.py for the line that redirects to the frontpage view and copy that to your own urls.py then copy over the index view and tweak it.

**Reply by p1-dta** [reply]
I have a HTTP ERROR 500.

I'm sorry, but even with https://docs.djangoproject.com/en/dev/ref/urls/#django.conf.urls.url
I don't understand what I have to write here:

```python
"""
Url definition file to redistribute incoming URL requests to django
views. Search the Django documentation for "URL dispatcher" for more
help.

"""
from django.conf.urls import url
# default evennia patterns
from evennia.web.urls import urlpatterns

# eventual custom patterns
from web.website import views

custom_patterns = [
    # url(r'index/', views),
    url(r'index/', views, name='example')
    # url(r'/desired/url/', view, name='example'),
]

# this is required by Django.
urlpatterns = custom_patterns + urlpatterns
```

**Reply by p1-dta** [reply]
With this, 

```python
# eventual custom patterns
from web.website.views import EvenniaIndexView

custom_patterns = [
    # url(r'index/', views),
    url(r'', EvenniaIndexView, name='example')
    # url(r'/desired/url/', view, name='example'),
]
```

I get a better error 

![image](https://user-images.githubusercontent.com/9381120/118413966-498f0900-b6a2-11eb-9060-deb7d62db7c5.png)

**Comment by Griatch** [comment, marked as answer]
- The `url()` call's first argument is a regex pointing to the index url. You find the default here: https://github.com/evennia/evennia/blob/master/evennia/web/website/urls.py#L12. 
- The second argument to `url()` is the _view_ (which is a Python function or callable class) for Django to call in order to generate a suitable HTML page for you to view. [here](https://github.com/evennia/evennia/blob/master/evennia/web/website/views.py#L110) is the view the default url points to. Copy this (and the `_gamestats` above it as well as any needed imports) into a new file in `mygame/web/`, let's call it `indexview.py`. 
- You then import your `indexview` into your `mygame/urls.py` and add the EvenniaIndexView to the `url()` call (so now it points to your local version).
- Reload and check that it works. Set `DEBUG=True` in your settings file to see proper tracebacks in your browser and not just a 500 error.
- Once that works, you can tweak your copy of the view as you like.

**Reply by p1-dta** [reply]
Ok I understand, thank you for your help, it works !
